src/app/user/login/login.component.html

```diff
@@ -1 +1,43 @@
-<p>login works!</p>
+<div class="form-group">
+  <form [formGroup]="loginForm" (ngSubmit)="onSubmit()">
+    <div
+      [ngClass]="{'is-invalid': isSubmitted && loginForm.get('email')!.errors && (loginForm.get('email')!.touched || loginForm.get('email')!.dirty)}">
+      <label for="email">Email</label>
+      <input type="email" id="email"
+             formControlName="email"
+             placeholder="Email-Id"
+      />
+      <span class="invalid-feedback"
+            *ngIf="isSubmitted && loginForm.get('email')!.errors && (loginForm.get('email')!.touched || loginForm.get('email')!.dirty)">
+        <span *ngIf="loginForm.get('email')!.errors?.required">
+          Email is required
+        </span>
+      </span>
+    </div>
+
+    <div
+      [ngClass]="{'is-invalid': isSubmitted && loginForm.get('password')!.errors && (loginForm.get('password')!.touched || loginForm.get('password')!.dirty)}">
+      <label for="password">Password</label>
+      <input type="password" id="password"
+             formControlName="password"
+             placeholder="Password"
+      />
+      <span class="invalid-feedback"
+            *ngIf="isSubmitted && loginForm.get('password')!.errors && (loginForm.get('password')!.touched || loginForm.get('password')!.dirty)">
+        <span *ngIf="loginForm.get('password')!.errors?.required">
```

Comment:
can you try to check the error type in the ts and then set the error message using a variable? this will avoid multiple rewrite of errors